packages/ragbits-core/src/ragbits/core/utils/config_handling.py

```diff
@@ -32,7 +36,62 @@ def get_cls_from_config(cls_path: str, default_module: ModuleType) -> Any:  # no
         except AttributeError as err:
             raise InvalidConfigError(f"Class {object_stringified} not found in module {module_stringified}") from err
 
+    if default_module is None:
+        raise InvalidConfigError("Given type string does not contain a module and no default module provided")
+
     try:
         return getattr(default_module, cls_path)
     except AttributeError as err:
         raise InvalidConfigError(f"Class {cls_path} not found in module {default_module}") from err
+
+
+class ObjectContructionConfig(BaseModel):
+    """
+    A model for object construction configuration.
+    """
+
+    # Path to the class to be constructed
+    type: str
+
+    # Configuration details for the class
+    config: dict[str, Any] = {}
+
+
+class WithConstructionConfig(abc.ABC):
+    """
+    A mixin class that provides methods for initializing classes from configuration.
+    """
+
+    # The default module to search for the subclass if no specific module is provided in the type string.
+    default_module: ClassVar[ModuleType | None] = None
+
+    @classmethod
+    def subclass_from_config(cls, config: ObjectContructionConfig) -> Self:
+        """
+        Initializes the class with the provided configuration. May return a subclass of the class,
+        if requested by the configuration.
+
+        Args:
+            config: A model containing configuration details for the class.
+
+        Returns:
+            An instance of the class initialized with the provided configuration.
+        """
+        subclass = get_cls_from_config(config.type, cls.default_module)
+        if not issubclass(subclass, cls):
+            raise InvalidConfigError(f"{subclass} is not a subclass of {cls}")
+
+        return subclass.from_config(config.config)
+
+    @classmethod
+    def from_config(cls, config: dict) -> Self:
```

Comment:
maybe classes could have some (optional) class variable `config_model` or something and this should be a default behavior to validate the config here ?

Reply:
Would you add a ticket about that to the backlog? I think it's an interesting idea to explore during grooming. But for now, I would leave that part as it was before, since the main purpose of this PR is to enable the work on CLI and the changes that are in this PR are enough for that.

Reply:
added to backlog: https://github.com/deepsense-ai/ragbits/issues/237